### PR TITLE
Show all donations for an event

### PR DESCRIPTION
**Pull request by alahab2**
This feature lets a User see all `Donations` for an `Event` after clicking in the `Event` name on `events/index` page.

Note! This change edited migrations so that an `Event` is created before a `Donation`. Please run: **rake db:drop db:create db:migrate db:seed** after pulling the changes.

See screenshots below:

screencapture-localhost-3000-events-4-1476199710597

---

**Before submitting, check that:**

 - [x] You have added (passing) tests for your code.
 - [x] You have written good* commit messages.
 - [x] You have squashed relevant commits together.
 - [x] You have ensured that RuboCop is passing.
 - [x] Your PR relates to one subject, with a clear title and
       description using complete, grammatically correct sentences.
---

If your change contains views, ensure that:

 - [x] You have included a screenshot of the new view.

---

*If you do not know what makes a good commit message, or why it
is important, please refer to these resources: [1][1], [2][2], [3][3].

[1]: https://robots.thoughtbot.com/5-useful-tips-for-a-better-commit-message
[2]: http://chris.beams.io/posts/git-commit/
[3]: https://github.com/blog/1943-how-to-write-the-perfect-pull-request

### CODE REVIEW

File: app/models/total_donations.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module TotalDonations
```
Comment (Drenmi) [inline app/models/total_donations.rb]: It's awesome that you've found mixins, and how we can use them to DRY up our code. I think it is a bit early to make a judgement about whether this abstraction is useful or not, so I would hold off on adding it. In general, I am much more comfortable keeping some duplicated code in there than I am adding abstractions too early. 😀

File: app/views/events/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        <th>Name</th>
+        <th>Date</th>
+        <th>Total Donations</th>
+      </tr>
+    </thead>
+    <tbody>
+      <% @event.donations.each do |donation| %>
+        <tr>
+          <td><%= donation.donor.name %></td>
+          <td><%= l donation.created_at.to_date, format: :default %></td>
```
Comment (Drenmi) [inline app/views/events/show.html.erb]: Is `format: :default` needed? 😀

File: config/routes.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 Rails.application.routes.draw do
   resources :donations, only: [:create, :update, :edit]
   resources :donors, only: [:show, :index, :update, :edit]
-  resources :events, only: [:index, :create]
+  resources :events, only: [:index, :create, :show]
```
Comment (Drenmi) [inline config/routes.rb]: Let's pick an arbitrary order for our actions, and use that order for all our routes. 😀

File: spec/controllers/events_controller_spec.rb [hunk 4]
```diff
@@ -48,4 +49,13 @@
       it { expect(response).to have_http_status(:ok) }
     end
   end
+
+  describe "GET #show" do
```
Comment (Drenmi) [inline spec/controllers/events_controller_spec.rb]: Awesome test!

File: spec/factories/events.rb [hunk 5]
```diff
@@ -7,5 +7,11 @@
     trait :invalid do
       name nil
     end
+
+    factory :event_with_donations do
+      after(:create) do |event, _|
```
Comment (Drenmi) [inline spec/factories/events.rb]: Not sure if the second argument can be omitted altogether? 😀

**Comment by alahab2** [comment]
All comments addressed 😎

**Comment by Drenmi** [comment]
🚀

**Pull request merged**